arch/arm/include/tls.h

```diff
@@ -37,12 +37,12 @@
  ****************************************************************************/
 
 /****************************************************************************
- * Name: up_getsp
+ * Name: arm_getsp
```

Comment:
Actually, in this case, up_getsp() is the correct name since it is a common interface that is exported outside of the ARM architecture.  Same is true of other locations.

Reply:
The function is in a header file that's exported but I don't think it will be used outside the ARM directory.

Reply:
It is an internal private function and not part of the well-defined interface, so it really is not important.

It is exported outside of ARM, however, if any code includes:

#include <arch/tls.h>

This it will get this code.  That could be from anywhere in the OS or even in applications.

But since it is not a part of the standard interface and since its prototype differs for each architecture, I don't think that the up_ naming is required.

Reply:
Maybe I can do something similar as RISC-V for instance and take the arm_getsp to the arch.h file.

Reply:
> Maybe I can do something similar as RISC-V for instance and take the arm_getsp to the arch.h file.

Don't bother.  It would still be exported to all files and should still have the up_ prefix.  There is no way to avoid avoid that.  TLS requires that that function be used outside of the ARM code.  Same is true for all of the other architectures.

I don't recommend changing the name, but not for that reason.  I don't recommend changing the because it is not a documented, controlled interface.  If is private for use within the tls.h header file.   it does not have a consist protoype, it is not prototyped in include/nuttx/arch.h so it doesn't matter what you call it.

up_ would be more arhicturually correct probably, but it is not important.  All MCU names used outide of the architecutre-/board-specific logic should be called up_